lightning/src/routing/scoring.rs

```diff
@@ -655,6 +668,10 @@ impl<G: Deref<Target = NetworkGraph<L>>, L: Deref, T: Time> Score for Probabilis
 	fn channel_penalty_msat(
 		&self, short_channel_id: u64, source: &NodeId, target: &NodeId, usage: ChannelUsage
 	) -> u64 {
+		if self.banned_nodes.contains(source) || self.banned_nodes.contains(target) {
+			return u64::max_value();
+		}
```

Comment:
Could you add a unit test?

Reply:
Will do!

Reply:
Added with https://github.com/lightningdevkit/rust-lightning/pull/1550/commits/c1671a5c90a66324d025268c069b502cb6ed97ce.